ocaml/parsing/jane_syntax.mli

```diff
@@ -121,6 +121,28 @@ module Strengthen : sig
     -> Parsetree.module_type_desc With_attributes.t
 end
 
+(** The ASTs for layouts. *)
+module Layouts : sig
+  type nonrec core_type =
+    (* ['a : immediate] or [_ : float64] *)
+    | Ltyp_var of { name : string option
+                  ; layout : Asttypes.layout_annotation }
+
+    (* [('a : immediate) 'b 'c ('d : value). 'a -> 'b -> 'c -> 'd] *)
+    | Ltyp_poly of { bound_vars : (string Location.loc *
+                                   Asttypes.layout_annotation option) list
+                   ; inner_type : Parsetree.core_type }
```

Comment:
I wondered the same thing. It's a bit awkward having them both. In the end, I decided "no", mostly because this takes up more space -- especially when you consider its encoding via attributes.

Reply:
I think I'd like a comment stating how it relates to `Ptyp_poly`, i.e. when it'll show up and what needs to handle it